### DISCUSSION TITLE
DS 26C32 ATM line driver and PCB-Design for measuring microscope

### DISCUSSION THREAD

**Discussion by Thomi3D** [MrGreensWorkshop/RasPiPicoSDK_PicoDRO#5] [How to]
Thank you very much for this great project, @MrGreensWorkshop.

My goal is to retrofit an old digital measuring microscope that has three RS422 glass scales with this DRO and the Raspberry Pi HQ camera. For this purpose, I want to change the DS8923 line driver to the DS 26C32 ATM, because of its availability in Germany. I would also like to design a PCB so that everything is nice and clean. Later on, my goal is to connect the Pico to a Raspberry Pi 4 that handles the camera.

Can you help me with this?

**Comment by MrGreensWorkshop** [comment]
Hi, I'm glad to hear that. Thanks for reaching out.
I think I mentioned in the video that the [DS8923](https://media.digikey.com/pdf/Data%20Sheets/Texas%20Instruments%20PDFs/DS8922,22A,23A.pdf) is a 5V device, but when I use a 5V supply, it gets too hot. I connected to a 3.3V rail to reduce current consumption and heat. Since I connected to a 3.3V rail, I didn't need to use a voltage level shifter between DS8923 and RpiPico.

[DS26C32AT](https://www.ti.com/lit/ds/symlink/ds26c32at.pdf) is also a 5V device. It should be connected to 5V. 
The datasheet says the minimum supply voltage is 4.5V,  which means they tested until 4.5V. But that doesn't mean it cannot work below 4.5V.

You have two choices:
1. You can test with a 3.3V supply, and if it works, you can go with it without using a level shifter. (not recommended. Maybe only for test purposes.)
1. Or you can connect to a 5V supply and use a voltage level shifter between DS26C32AT and RpiPico.

The voltage level shifter doesn't need to be bidirectional. You may also use a fast-switching MOSFET or transistor for voltage level shifting. Since there are three axes, you may want to use a level shifter IC to reduce components.
I will check both datasheets and get back to you.

**Reply by Thomi3D** [reply]
Thank you very much for the answer.

What do you think about the AM26LV32CD, which can operate at 3.3 V while accepting 5 V logic signals and only consumes about 27 mW? This would also be available at my dealership and i would not need the level shifting. 

Greetings from Germany
Thomas

**Reply by MrGreensWorkshop** [reply]
Hi, Thomas. This is even better. Yes, you don't need a level shifter with [AM26LV32CD](https://www.ti.com/lit/ds/symlink/am26lv32.pdf).
I will check for the required connection to replace it with [DS8923](https://media.digikey.com/pdf/Data%20Sheets/Texas%20Instruments%20PDFs/DS8922,22A,23A.pdf) and get back to you.

**Reply by Thomi3D** [reply]
Hello @MrGreensWorkshop, I've studied the datasheet of the AM26LV32CD a bit, and with my limited knowledge, it seems that the failsafe circuitry is no longer needed because it's already integrated into the IC. Also, the unused input pins can remain open, which makes things even easier. As a mechanical engineer with limited compotence in this area, this looks very promising to me. What do you think about this?

Greetings 
Thomas

**Reply by MrGreensWorkshop** [reply]
Hi, Thomas
>I've studied the datasheet of the AM26LV32CD a bit, and with my limited knowledge, it seems that the failsafe circuitry is no longer needed because it's already integrated into the IC.

I didn't check properly, but on which page did you get that impression? Could you be more specific so I can check later on?

>the unused input pins can remain open, which makes things even easier.

Yes, you are right. I checked that at page 14.

**Reply by Thomi3D** [reply]
Hi @MrGreensWorkshop,

on page 11 (chapter 8.4.1: Fail-Safe Conditions) it says: "The AM26LV32 device has an internal fail-safe circuitry which prevents the device from putting an 
unknown voltage signal at the receiver outputs." 

Hope this helps. 

Greetings
Thomas

**Reply by MrGreensWorkshop** [reply]
Hi, About the failsafe circuit, you may be right. But a RS422 termination resistor is still needed.
Why don't you try to replace one of the driver circuits on the schematic and let me check?
You can use Paint. No schematic drawing software is needed at this stage. (You can also draw on paper and take a picture of it.)
What we do is just circuit design review.

**Reply by Thomi3D** [reply]
Hello @MrGreensWorkshop,

thanks again for your effort. 
I will try to design a circuit when i find the time and get back to you. 

Greetings
Thomas

**Reply by Thomi3D** [reply]
Hello @MrGreensWorkshop,

this is what i came up with: 
![Pico-DRO](https://github.com/MrGreensWorkshop/RasPiPicoSDK_PicoDRO/assets/164365976/da7a80bc-e40c-40e7-9180-731e1e15bc40)


I hope it's not completely wrong, and I'm looking forward to your feedback. :)

Greetings
Thomas

**Reply by MrGreensWorkshop** [reply]
It looks good to me. 
Pin 26 is used for A input, and pin 27 is used for B input. It is reversed, but not a big deal. 
One more thing. I don't know how many drivers you are going to use, but please don't forget to check the Raspberry Pi Pico's 3.3V regulator datasheet to make sure it can supply enough power.

**Reply by Thomi3D** [reply]
Thank you for your feedback. The Picos 3.3V has a maximum output of 300mA, which should be sufficient for the three drivers I'm planning to use. I will order the parts now and will keep you updated on the progress. :)

Best regards,
Thomas

**Reply by MrGreensWorkshop** [reply]
No problem at all. Thank you for bringing up AM26LV32CD.
Good luck!
